apps/web/pages/api/intercom-hash.ts

```diff
@@ -0,0 +1,28 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+import crypto from "node:crypto";
+import { z } from "zod";
+
+export default function handler(req: NextApiRequest, res: NextApiResponse) {
+  const { query, method } = req;
```

Comment:
This endpoint seems open and anyone can send any user ID. Shouldn't we restrict it to userId of the session? In that case, `userId` can be derived from session itself.